server.go

```diff
@@ -204,9 +204,9 @@ func (s *Server) fsScanner(done <-chan void) {
 				return
 			default:
 			}
-			fileCacheMtx.Lock()
+			fileCacheMtx.RLock()
 			fcCopy := fileCache
```

Comment:
Not your fault, but this is still crap. Maps are effectively pointers, so this shallow copy doesn't solve anything.

Reply:
Can I use [copy](https://golang.org/pkg/builtin/#copy) instead?